### PR TITLE
Add option toggle autoAddInvalidUnusual = false

### PR DESCRIPTION
**Pull request by joekiller**
If the bot accepts an Unusual item that is not on the price list it will not put it back up for sale unless autoAddInvalidUnusual is set to true. The option is set to false by default to protect users who use the generic price SKU and unaware that autoAddInvalidItems would sell off anything purchased via that generic sku.

Based off conversation in #400

### CODE REVIEW

File: src/classes/MyHandler/offer/accepted/updateListings.ts [hunk 1]
```diff
@@ -44,6 +44,9 @@ export default function updateListings(
         const isDisabledHV = highValue.isDisableSKU.includes(sku);
         const isAdmin = bot.isAdmin(offer.partner);
         const isNotSkinsOrWarPaint = SKU.fromString(sku).wear === null;
+        // if item is unusual and autoAddInvalidUnusual is set to true then we allow addInvalidUnusual.
+        // If the item is not an unusual sku, we "allow" still (no-op)
+        const addInvalidUnusual = SKU.fromString(sku).quality === 5 ? opt.pricelist.autoAddInvalidUnusual : true;
```
Comment (idinium96) [inline src/classes/MyHandler/offer/accepted/updateListings.ts:49]: This is mainly for the Generic Unusual feature right?
So I guess it's much better to do it like this (together with some small refactorization):

```ts
const item = SKU.fromString(sku);
const isNotSkinsOrWarPaint = item.wear === null;
const addInvalidUnusual = item.quality === 5 && item.effect === null ? opt.pricelist.autoAddInvalidUnusual : true;
```

File: src/classes/Options.ts [hunk 2]
```diff
@@ -83,6 +83,9 @@ export const DEFAULTS = {
         autoAddInvalidItems: {
             enable: true
         },
+        addInvalidUnusual: {
```
Comment (idinium96) [inline src/classes/Options.ts:86]: rename this to `autoAddInvalidUnusual` will fix the test error.

**Review by idinium96 — CHANGES_REQUESTED** [review]

File: src/classes/MyHandler/offer/accepted/updateListings.ts (hunk 1 above)
Comment (idinium96) [reply src/classes/MyHandler/offer/accepted/updateListings.ts:49]: oh, wait. I don't think `item.effect === null` is correct.

Comment (idinium96) [reply src/classes/MyHandler/offer/accepted/updateListings.ts:49]: Yeah, scratch that. My bad.

Comment (joekiller) [reply src/classes/MyHandler/offer/accepted/updateListings.ts:49]: yeah I was about to say. I think in this case the SKU detected would be the SKU of the actual item at this point, so it wuld have an effect

Comment (idinium96) [reply src/classes/MyHandler/offer/accepted/updateListings.ts]: Yep correct. It's not like when receiving the painted item (with `normalize.painted.their` set to `false`)

File: src/classes/MyHandler/offer/accepted/updateListings.ts [hunk 3]
```diff
@@ -36,14 +36,18 @@ export default function updateListings(
...
         const inPrice = bot.pricelist.getPrice(sku, false);
 
         const existInPricelist = inPrice !== null;
         const isDisabledHV = highValue.isDisableSKU.includes(sku);
         const isAdmin = bot.isAdmin(offer.partner);
-        const isNotSkinsOrWarPaint = SKU.fromString(sku).wear === null;
+        const isNotSkinsOrWarPaint = item.wear === null;
+        // if item is unusual and autoAddInvalidUnusual is set to true then we allow addInvalidUnusual.
+        // If the item is not an unusual sku, we "allow" still (no-op)
+        const addInvalidUnusual = item.quality === 5 ? opt.pricelist.autoAddInvalidUnusual : true;
```
Comment (joekiller) [inline src/classes/MyHandler/offer/accepted/updateListings.ts:50]: this is somewhat a strange statement but it really should be "true" if either:
1. the item is unsual and we autoAddInvalidUnusual or,
2. if the item isn't an unusual.

File: src/classes/MyHandler/offer/accepted/updateListings.ts (hunk 1 above)
Comment (joekiller) [reply src/classes/MyHandler/offer/accepted/updateListings.ts]: this is somewhat a strange statement but it really should be "true" if either:

the item is unsual and we autoAddInvalidUnusual or,
if the item isn't an unusual.

**Review by idinium96 — COMMENTED** [review]
> this is somewhat a strange statement but it really should be "true" if either:

> - the item is unsual and we autoAddInvalidUnusual or,
> - if the item isn't an unusual.

very true. but will the bot notify the owner about this?

**Comment by joekiller** [comment]
Not at the moment. That could be considered a secondary part to this safety net. An option like notifyAcceptedInvalidUnusual or something?

**Comment by idinium96** [comment]
yeah, something like that. I might add that while you're focusing on your medical work.

**Comment by idinium96** [comment]
The latest commit should do the notifying job.

**Comment by idinium96** [comment]
this looks good now I guess. merging...

**Pull request merged**